plugin.video.hbogoeu/hbogolib/ttml2srt.py

```diff
@@ -0,0 +1,199 @@
+# encoding: utf-8
+#
+#  --------------------------------------------
+#  based on https://github.com/yuppity/ttml2srt
+#  --------------------------------------------
+
+from __future__ import absolute_import, division
+
+import sys
+
+from defusedxml import minidom
+
+
+class Ttml2srt(object):
+
+    def __init__(self, ttml_input_file, default_fps):
+        self.subtitle = self.extract_subtitle_data(ttml_input_file)
+        if not self.subtitle['fps']:
+            self.subtitle['fps'] = default_fps
+
+    def write_srt_file(self, str_file, shift=0):
+        filename = str_file + '.srt'
+        if self.subtitle['lang'] is not None:
+            filename = str_file + '.' + self.subtitle['lang'] + '.srt'
+        f = None
+        if (sys.version_info > (3, 0)):
+            f = open(filename, 'w', encoding='utf-8')
```

Comment:
xbmcvfs maybe?

Reply:
I prefer to keep this as it is so I can test it automatically with external tools outside of Kodi. Would the use of xbmcvfs provide an advantage here?

Reply:
Only if the file is on network shares since NFS and SMB shares are transparent to xbmcvfs

Reply:
Ok tnx, the subtitles are read and written to/from the userdata/addon_data/... folder only